### ISSUE TITLE
Support using multiple -config.file CLI parameter for promtail

### ISSUE THREAD

**Issue by Mario-Hofstaetter** [grafana/loki#5802]
**Is your feature request related to a problem? Please describe.**

TL;DR: When using `-config.file` CLI parameter twice, the config from first file is discarded / replaced by promtail default config values.
Only settings existing in second file should override previous settings.

----

We are using promtail on windows virtual machines to gather all kinds of logs.
Providing users with a default config for running promtail is straightforward using a config-file, which is version-controlled.

However, some settings MUST be changed by users to something different than is committed in git / setup backups als default.

Best example is `url` of [`client`](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#clients) to point to correct loki server.

I tried using the CLI parameter twice, first with the default config followed by a minimal config file containing only user set override values (see config below) like this:

```text
.\promtail.exe "-config.file" .\promtail-config.yaml "-config.file" .\promtail-config.override.yaml
```

However, the second parameter resets all config values not present in `promtail-config.override.yaml` to promtail defaults.

Since we have to deal with windows and other plattform limitations, recommended practises like config-management tools etc. are not feasible. Promtail is set up in offline / air gapped environments by sometimes less tech savvy users.

So a second config file for user provided values would probably be the must user friendly.

**Describe the solution you'd like**

When using `-config.file` parameter multiple times, the configuration should be *merged* in order of arguments and then applied alltogether (config below).

For example this is possible in https://opentelemetry.io/docs/collector/

**Describe alternatives you've considered**

### CLI Flags

Using CLI flag `-client-url` could be possible, however I ran into unexpected behavior. Running:

```
.\promtail.exe "-config.file" .\promtail-config.yaml "-client.url" https://ax-monitor-02.ax.tech/loki/loki/api/v1/push "-config.expand-env=true"
```

yielded TWO client configs in promtail. I expected the CLI flag to overwrite URL of client present in config file, but got this:

(default properties omitted for brevity):

```text
client:
  url: https://SomeOtherServerThanDefault/loki/api/v1/push

clients:
- url: https://DefaultServerFromConfigFile/loki/api/v1/push
  external_labels: |
    lokiclient: promtail
    instance: MYMACHINE:9080
    job: MYMACHINE
```

The `client` is also missing `external_labels` because that CLI flag was not provided.

How can I overwrite the existing first and only client in `clients` section from config file?

### ENV Vars

Another possibilty is using ENV variables since we are already using `-config.expand-env=true`.
However we need to run `promtail` as a windows service which makes things cumbersome.

**Additional context**

Config files used:

<details>
<summary>promtail-config.yaml (click me)</summary>

```yaml
server:
  http_listen_address: 127.0.0.1
  http_listen_port: 9080
  grpc_listen_port: 0

clients:
  - url: https://DefaultServerFromConfigFile/loki/api/v1/push
    external_labels: 
      instance: "${COMPUTERNAME}:9080"
      job: "${COMPUTERNAME}"

positions:
  filename: ./tmp/positions.yaml

scrape_configs:
    # etc. !
```
</details>

<details>
<summary>promtail-config.override.yaml (click me)</summary>

This should only replace this section in previous `promtail-config.yaml`

```yaml
clients:
  - url: https://SomeOtherServerThanDefault/loki/loki/api/v1/push
    external_labels: 
      instance: "${COMPUTERNAME}:9080"
      job: "${COMPUTERNAME}"
```
</details>

**Comment by thapakazi** [comment]
Anyone working on directory support  option: `-config.dir ...` ?

**Comment by cstyan** [comment]
Hello, thanks for your feature request!

We're currently reevaluating promtails position as a project within Grafana Labs. Internally we're actually using the [Agent](https://github.com/grafana/agent) for both metrics and logs collection at this point. In addition, the agent already supports combining of config files via it's Flow configuration model.

While we haven't made a formal decision yet, we expect in the near future that all new feature work will be done in the Agent's log collection pipelines rather than in Promtail.

**Comment by JStickler** [comment]
Promtail is end of life (EOL) as of March 2, 2026. No future support or updates will be provided. All future feature development will occur in [Grafana Alloy](https://github.com/grafana/alloy).
